EIPS/eip-draft_universal_router.md

```diff
@@ -0,0 +1,282 @@
+---
+eip: <to be assigned>
+title: Universal Router Contract
+description: Universal router contract designed for token allowance that eliminates all `approve` transactions in the future.
+author: Zergity <zergity@gmail.com>
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-12-12
+requires: EIP20, EIP721, EIP1155
+---
+
+## Abstract
+
+A universal router contract that executes transactions with the following steps:
+  * (optional) call a calculation contract to get the `amountIn` value, and ensure that this `amountIn` is not greater than an input `amountInMax`
+  * transfer `amountIn` of a token from `msg.sender` to a `recipient` address
+  * call a contract function to execute an action
+  * (optional) verify the returning amount of a token must be no less than an input `amountOutMin`
```

Comment:
This is probably too detailed for the abstract, and might be better in the specification section. Try to give a high level overview, while still remaining technical.